app/src/main/java/com/moonpig/mvisample/di/productdetail/ProductDetailsModule.kt

```diff
@@ -0,0 +1,27 @@
+package com.moonpig.mvisample.di.productdetail
+
+import com.moonpig.mvisample.domain.productdetail.ProductDetailRepository
+import com.moonpig.mvisample.domain.productdetail.ProductDetailUseCase
+import com.moonpig.mvisample.productdetail.ProductDetailTracker
+import com.moonpig.mvisample.productdetail.ProductDetailViewModelFactory
+import dagger.Module
+import dagger.Provides
+
+@Module
+class ProductDetailsModule {
+
+    @Provides
+    fun provideProductDetailsUseCase(productDetailsRepository: ProductDetailRepository) =
+            ProductDetailUseCase(productDetailsRepository)
+
+    @Provides
+    fun provideProductDetailTracker(): ProductDetailTracker =
+            ProductDetailTracker()
+
+    @Provides
+    fun provideProductDetailsViewModelFactory(
```

Comment:
There's an alternative way of doing this so you don't need to specify the viewModelFactory in every module. See `com.moonpig.di.viewmodel` package in the Moonpig project

Reply:
We decided that for MVI demonstration purposes this is not necessary right now. 
Me might want to add more MVI activities in the future and then add the `DaggerViewModelFactory` but in the current state it could be to complex and confusing to someone who just wants to understand MVI